From 25d46c4b6c334e570c25a318ba77876883516c6e Mon Sep 17 00:00:00 2001
From: Zumeng Chen <zumeng.chen@windriver.com>
Date: Mon, 25 Sep 2017 14:01:34 +0800
Subject: [PATCH 1/1] km: 4.8 for fixing compiling error

services4/srvkm/env/linux/osfunc.c:3703:118: error: passing argument 6 of 'get_user_pages_remote' makes integer from pointer without a cast [-Werror=int-conversion]
     psInfo->iNumPagesMapped = get_user_pages_remote(current, current->mm, uStartAddr, psInfo->iNumPages, FOLL_WRITE, psInfo->ppsPages, NULL);
                                                                                                                      ^~~~~~
In file included from ./arch/arm/include/asm/cacheflush.h:13:0,
                 from services4/srvkm/env/linux/osfunc.c:55:
./include/linux/mm.h:1311:6: note: expected 'int' but argument is of type 'struct page **'
 long get_user_pages_remote(struct task_struct *tsk, struct mm_struct *mm,
      ^~~~~~~~~~~~~~~~~~~~~
services4/srvkm/env/linux/osfunc.c:3703:31: error: too few arguments to function 'get_user_pages_remote'
     psInfo->iNumPagesMapped = get_user_pages_remote(current, current->mm, uStartAddr, psInfo->iNumPages, FOLL_WRITE, psInfo->ppsPages, NULL);
                               ^~~~~~~~~~~~~~~~~~~~~
In file included from ./arch/arm/include/asm/cacheflush.h:13:0,
                 from services4/srvkm/env/linux/osfunc.c:55:
./include/linux/mm.h:1311:6: note: declared here
 long get_user_pages_remote(struct task_struct *tsk, struct mm_struct *mm,
      ^~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors

Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 eurasia_km/services4/srvkm/env/linux/osfunc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/eurasia_km/services4/srvkm/env/linux/osfunc.c b/eurasia_km/services4/srvkm/env/linux/osfunc.c
index d3696fa..71f53dd 100644
--- a/eurasia_km/services4/srvkm/env/linux/osfunc.c
+++ b/eurasia_km/services4/srvkm/env/linux/osfunc.c
@@ -3700,7 +3700,7 @@ PVRSRV_ERROR OSAcquirePhysPageAddr(IMG_VOID *pvCPUVAddr,
     bMMapSemHeld = IMG_TRUE;
 
     /* Get page list */
-    psInfo->iNumPagesMapped = get_user_pages_remote(current, current->mm, uStartAddr, psInfo->iNumPages, FOLL_WRITE, psInfo->ppsPages, NULL);
+    psInfo->iNumPagesMapped = get_user_pages_remote(current, current->mm, uStartAddr, psInfo->iNumPages, FOLL_WRITE,0, psInfo->ppsPages, NULL);
 
     if (psInfo->iNumPagesMapped >= 0)
     {
-- 
2.7.4

